api/versions.go

```diff
@@ -219,3 +221,43 @@ func (self *version1_2) HandleRequest(requestType string, request []string, m ma
 		return self.baseVersion.HandleRequest(requestType, request, m, w, r)
 	}
 }
+
+// v2.0 builds on v1.2
+type version2_0 struct {
+	baseVersion *version1_2
+}
+
+func newVersion2_0(v *version1_2) *version2_0 {
```

Comment:
I think v2.0 shouldn't be based on v1.X, it'll probably be separate completely.

Maybe we keep this for now since it is not complete though. Although having an incomplete WIP API will probably discourage others from using it until it is ready :)